### ISSUE TITLE
Would be nice to support directly loading TREC run and qrel files directly

### ISSUE THREAD

**Issue by cmacdonald** [cvangysel/pytrec_eval#5]
or additionally, dataframe of the same format.

**Comment by cvangysel** [comment]
Do you mean like the following?
- https://github.com/cvangysel/pytrec_eval/blob/master/py/__init__.py#L16
- https://github.com/cvangysel/pytrec_eval/blob/master/py/__init__.py#L28
- https://github.com/cvangysel/pytrec_eval/blob/master/examples/trec_eval.py#L10

**Comment by cmacdonald** [comment]
Yes, I realised that after. Documenting these somewhere would help. Dataframe API would be good though :-)

**Comment by plopezgarcia** [comment]
Hi everyone. I think I might have already done part of the job.

This year I participated in the [Precision Medicine track](http://www.trec-cds.org/2018.html) I wrote some [small utilities](https://github.com/plopezgarcia/trec-2018-precision-medicine/tree/master/python/trec_utils) on top of pytrec_eval that use dataframes as the main data structure and deal with little issues like:

- [Loading a qrels file as a dataframe](https://github.com/plopezgarcia/trec-2018-precision-medicine/blob/master/python/trec_utils/utils.py#L70)
- [Convert that dataframe into the qrels dict format of pytrec_eval](https://github.com/plopezgarcia/trec-2018-precision-medicine/blob/master/python/trec_utils/utils.py#L95)
- [Dump a dataframe containing a run into trec_eval format](https://github.com/plopezgarcia/trec-2018-precision-medicine/blob/master/python/trec_utils/utils.py#L116)

You can find a full example of usage here [here](https://github.com/plopezgarcia/trec-2018-precision-medicine/blob/master/python/abstracts/abstracts_submissions.ipynb).

Some things are still dependent on my code, but I think most of it could be generalized and reused? Let me know if I can help.
